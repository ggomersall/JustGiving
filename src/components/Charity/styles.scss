@import '../BaseStyles/mixins';
@import '../BaseStyles/variables';

.charity__section {
  background: $white;
}

.charity-content__wrapper {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 20px;
  height: 230px;
  .charity__logo img {
    width: 200px;
  }

  .charity-description__wrapper {
    .charity__title {
      margin-bottom: 20px;
    }
    .charity__statement {
      width: 400px;
    }
  }

  .charity-button__wrapper {
    display: flex;
    flex-direction: column;
    .btn {
      margin-bottom: 15px;
      font-family: 'Roboto',Arial,Helvetica,sans-serif;
      letter-spacing: 1px;
    }
  }

  .btn--orange {
    @include btn--orange;
  }
  .btn--purple {
    @include btn--purple;
  }

}