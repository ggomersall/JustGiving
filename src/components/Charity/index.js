import React from 'react';
import PropTypes from 'prop-types';
import Care from '../Care';

import './styles.scss';

const Charity = ({
  charity, likes, handleCareUpdate, care
}) => (
  <div className="charity__section">
    <div className="charity-content__wrapper container">
      <div className="charity__logo">
        <img src={charity.logoAbsoluteUrl} width="200" />
      </div>
      <div className="charity-description__wrapper">
        <h2 className="charity__title">{charity.name}</h2>
        <h3 className="charity__statement">{`We ${charity.impactStatementWhat} to ${charity.impactStatementWhy}`}</h3>
        <div className="charity__care">
          <Care handleCareUpdate={handleCareUpdate} likes={likes} care={care} />
        </div>
      </div>
      <div className="charity-button__wrapper">
        <button className="btn btn--orange">Fundraise for us</button>
        <button className="btn btn--purple">Donate</button>
      </div>
    </div>
  </div>
);

Charity.propTypes = {
  charity: PropTypes.object,
  likes: PropTypes.number,
  handleCareUpdate: PropTypes.func,
  care: PropTypes.bool
};

export default Charity;
