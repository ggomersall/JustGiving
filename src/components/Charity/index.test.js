import React from 'react';
import { shallow } from 'enzyme';
import Charity from './index.js';

describe('Testing Charity Component', () => {

  let wrapper;

  test('it should render data correctly', () => {
    const likes = 100;
    const care = true;
    const charity = { 
      logoAbsoluteUrl: 'https://placeholder.com/150',
      name: 'Macmillan', 
      impactStatementWhat: 'lorem test', 
      impactStatementWhy: 'extra lorem text' 
    };
    wrapper = shallow(<Charity charity={charity} likes={likes} care={care} />);
  });

  test('Snapshot', () => expect(<Charity />).toMatchSnapshot());

});

