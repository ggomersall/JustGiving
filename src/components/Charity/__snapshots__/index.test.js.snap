// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Charity Component Snapshot 1`] = `<Charity />`;
